src/SolutionAcceleration.f90

```diff
@@ -0,0 +1,395 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief A Fortran 2003 module that encompasses a base class for solution
+!>        acceleration.  Module also includes both picard iteration and a
+!>        modified picard iteration.  Anderson Acceleration extends this base
+!>        class in the AndersonAcceleration module.
+MODULE SolutionAccelerationModule
+#include "Futility_DBC.h"
+USE Futility_DBC
+USE IntrType
+USE FutilityComputingEnvironmentModule
+USE ParameterLists
+USE ParallelEnv
+USE VectorTypes
+
+IMPLICIT NONE
+
+PRIVATE
+
+!List of public members
+PUBLIC :: SolutionAccelerationType
+PUBLIC :: RelaxedPicardType
+PUBLIC :: ModifiedPicardType
+
+!> @brief the base Solution Acceleration Type
+TYPE,ABSTRACT :: SolutionAccelerationType
+  !> Initialization status
+  LOGICAL(SBK) :: isInit=.FALSE.
+  !> Size of solution vector/matrix
+  INTEGER(SIK) :: n=-1
+  !> Current iteration count
+  INTEGER(SIK) :: s=0
+  !> Starting iteration
+  INTEGER(SIK) :: start=1
+  !> Initial iterates
+  CLASS(VectorType),ALLOCATABLE :: x(:)
+  !> Futility computing environment
+  TYPE(FutilityComputingEnvironment),POINTER :: ce => NULL()
+!
+!List of Type Bound Procedures
+  CONTAINS
+    !> @copybrief SolutionAccelerationType::init
+    !> @copydetails SolutionAccelerationType::init
+    PROCEDURE(init_absintfc),DEFERRED,PASS :: init
+    !> @copybrief SolutionAccelerationType::init_base_SolutionAccelerationType
+    !> @copydetails SolutionAccelerationType::init_base_SolutionAccelerationType
+    PROCEDURE,PASS :: init_base => init_base_SolutionAccelerationType
+    !> @copybrief SolutionAccelerationType::clear
+    !> @copydetails SolutionAccelerationType::clear
+    PROCEDURE(clear_absintfc),DEFERRED,PASS :: clear
```

Comment:
You could actually do `PROCEDURE(clear_base_SolutionAccelerationType)` here and not define the abstract interface.  Reduces a few lines of code and basically says "the concrete clear routines need to have the same interface as the base clear routine"

You might be able to do the same thing with `init`, but I haven't gotten far enough down to see if that interface holds or not

Reply:
But if the base class doesn't have the interface defined, then we'll need select types everywhere to handle.  Or are you saying use clear and clear_base as the same?

Reply:
I'm saying you can pass an actual subroutine name into the `PROCEDURE(...)`.  You only need the abstract interface `clear_absintfc` if there's no routine in the module that defines the right interface.

Another option is to not have `clear` be deferred... just point it at `clear_base_SolutionAccelerationType` and let it get overridden later.  That's probably simpler actually